daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/RollbackTest.scala

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package speedy
+
+import com.daml.lf.PureCompiledPackages
+import com.daml.lf.data.ImmArray
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.language.Ast._
+import com.daml.lf.speedy.Compiler.FullStackTrace
+import com.daml.lf.speedy.SResult.SResultScenarioCommit
+import com.daml.lf.testing.parser.Implicits._
+import com.daml.lf.transaction.Node
+import com.daml.lf.transaction.SubmittedTransaction
+import com.daml.lf.validation.Validation
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value._
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AnyWordSpec
+
+class ExceptionTest extends AnyWordSpec with Matchers with TableDrivenPropertyChecks {
+
+  private def typeAndCompile(pkg: Package): PureCompiledPackages = {
+    val rawPkgs = Map(defaultParserParameters.defaultPackageId -> pkg)
+    Validation.checkPackage(rawPkgs, defaultParserParameters.defaultPackageId, pkg)
+    data.assertRight(
+      PureCompiledPackages(rawPkgs, Compiler.Config.Default.copy(stacktracing = FullStackTrace))
+    )
+  }
+
+  private def runUpdateExpr_GetTx(pkgs1: PureCompiledPackages)(e: Expr): SubmittedTransaction = {
```

Comment:
```suggestion
  private def runUpdateExprGetTx(pkgs1: PureCompiledPackages)(e: Expr): SubmittedTransaction = {
```
unless there is a good reason to deviate from Scala naming conventions, let’s stick to them.

Reply:
yup